frontend/src/components/ServerList.tsx

```diff
@@ -1,70 +1,134 @@
+                            }).catch(() => {
+                                setPlayerCounts((prevPlayerCounts) => {
+                                    return {
+                                    ...prevPlayerCounts,
+                                    [serverId]: 0
+                                    };
+                                });
+                            })
+                        } else {
+                            console.error("Parsing server key failed");
+                        }
+                    })
+                );
+            }
+        };
 
-                    const server = servers[index]; // Parse the key to a number
-                    if(serverListType === ServerListType.LIST){
-                        return (
-                            <ListItem key={index}>
-                                <ListItemButton onClick={() => {setActiveServer(index); setDrawerOpen(false)}}>
-                                    {index}: {server.serverAlias? server.serverAlias : "Unnamed Server"}
-                                </ListItemButton>
-                            </ListItem>
-                        );
-                    }
-                    else if(serverListType === ServerListType.CARD){
-                        console.log(server)
-                        return (
-                            <ListItem className={'w-[calc(100%_*_(1/4))] p-[10px]'} key={index}>
-                                <Card className={'cursor-pointer w-full'} onClick={() => {setActiveServer(index)}}>
-                                    <div className={'px-2'}>
-                                        <div className={'text-xl font-bold break-all'}>{server.serverAlias? server.serverAlias : "Unnamed Server"}</div>
-                                        <div className={'text-lg mb-2'}>{server.ipAddress}:{server.queryPort}</div>
+        refreshPlayerCounts();
+    }, [servers]);
+
+    useEffect(() => {
+        const refreshServerStatuses = async () => {
+            if (servers) {
+                await Promise.all(
+                    Object.keys(servers).map(async (key) => {
+                        const serverId = parseInt(key, 10);
+                        if (!isNaN(serverId)) {
+                            GetServerStatus(serverId).catch((reason) => {console.error("serverstatus: " + reason); addAlert(reason, "danger")}).then((count) => {
+                                setServerStatuses((prevServerStatuses) => {
+                                    return {
+                                        ...prevServerStatuses,
+                                        [serverId]: count as boolean | undefined
+                                    };
+                                });
+                            });
+                        } else {
+                            console.error("Parsing server key failed");
+                        }
+                    })
+                );
+            }
+        };
```

Comment:
Statements nest too deep, you can prevent this by for example doing if(!servers) return